### PR TITLE
Add basic CSP support

### PR DESCRIPTION
**Pull request by jelly**
Add Content Security Policy support using flask-talisman. By default
the force https and session_cookie_secure (forces cookies to be send over https)
options are enabled which break a normal `python run.py` on a local
machine, therefore they have been disabled.

CSP exceptions have been made for style, fonts served from external
source i.e. gooogle and form-action has been set as it's required for
form submission.

### CODE REVIEW

**Comment by anthraxx** [comment]
cross-referencing #40

**Comment by jelly** [comment]
Rebased the PR on master so that our CI checks run (which of course finds an issue)

**Comment by anthraxx** [comment]
travis doesn't like this yes, can you alter your previous commit and include the dependency in the .travis.yml file?

**Comment by jelly** [comment]
Just updated the .travis.yml file too.

File: app/__init__.py [hunk 1]
```diff
@@ -24,9 +25,17 @@ def set_sqlite_pragma(dbapi_connection, connection_record):
     dbapi_connection.isolation_level = isolation_level
 
 
+csp = {
+    'default-src': '\'self\'',
+    'style-src': '\'self\' ajax.googleapis.com fonts.googleapis.com ' '*.gstatic.com',
```
Comment (kpcyrd) [inline app/__init__.py]: I think this
```python
'style-src': '\'self\' ajax.googleapis.com fonts.googleapis.com ' '*.gstatic.com',
```
can be replaced with
```python
'style-src': '\'self\' ajax.googleapis.com fonts.googleapis.com *.gstatic.com',
```

**Comment by anthraxx** [comment]
we host own fonts now via 1885b0229ff1ad6be380111241daadd2e6ba5381 it would be neat if you could remove the google domains and only allow self.

**Pull request merged**